LoginClient/PipeClient.cs

```diff
@@ -0,0 +1,374 @@
+        /// </summary>
+        public delegate void UpdateTheFormHandler();
+
+        /// <summary>
+        /// The event that is called when the client connects to the server.
+        /// </summary>
+        public event UpdateTheFormHandler UpdateTheForm;
+
+        const int BUFFER_SIZE = 4096;
+
+        FileStream stream;
+        SafeFileHandle handle;
+        Thread readThread;
+
+        /// <summary>
+        /// Is the client connected to a server pipe.
+        /// </summary>
+        public bool isConnected { get; private set; }
+
+        /// <summary>
+        /// The name of the pipe connected to the server.
+        /// </summary>
+        public string pipeName { get; private set; }
+
+        /// <summary>
+        /// Is the client logged in as the admin.
+        /// </summary>
+        public bool isLoggedIn { get; private set; }
+
+        /// <summary>
+        /// The methods for accessing the 
+        /// connected, pipename, and isLoggedOn properties.
+        /// </summary>
+        /// <returns></returns>
+        #region Accessors
+
+        //public bool isLoggedIn()
+        //{
+        //    return isLoggedIn;
+        //}
+
+        //private void isLoggedIn(bool isLoggedInOrNot)
+        //{
+        //    isLoggedIn = isLoggedInOrNot;
+
+        //    if (isLoggedIn)
+        //    {
+        //        UpdateTheForm();
+        //    }
+        //}
+
+        /// <summary>
+        /// Returns true if client is connected to a server.
+        /// </summary>
+        /// <returns>Boolean</returns>
+        //public bool isConnected()
+        //{
+        //    return isConnected;
+        //}
+
+        /// <summary>
+        /// Set the value of the connected boolean.
+        /// </summary>
+        /// <param name="b">The new value of connected.</param>
+        //private void isConnected(bool b)
+        //{
+        //    isConnected = b;
+        //}
+
+        /// <summary>
+        /// Return the name of the pipe used to connect to the server.
+        /// </summary>
+        /// <returns>string</returns>
+        //public string pipeName()
+        //{
+        //    return pipeName;
+        //}
+
+        /// <summary>
+        /// Set the name of the pipe that connects to the server.
+        /// </summary>
+        /// <param name="newPipeName">The new pipe name.</param>
+        //private void pipeName(string newPipeName)
+        //{
+        //    pipeName = newPipeName;
+        //}
+
+        #endregion
+
+
+        #region Connection
+
+        /// <summary>
+        /// Disconnects the client from the server.
+        /// </summary>
+        public void Disconnect()
+        {
+            if(!isConnected)
+            {
+                return;
+            }
+
+            // We're no longer connected to the server.
+            isConnected = false;
+            pipeName = null;
+            isLoggedIn = false;
+            UpdateTheForm();
```

Comment:
Future Bug.
UpdateTheForm?.Invoke();
so you dont have to check if its assigned.